numpy/linalg/_linalg.py

```diff
@@ -815,6 +827,8 @@ def cholesky(a):
     with errstate(call=_raise_linalgerror_nonposdef, invalid='call',
                   over='ignore', divide='ignore', under='ignore'):
         r = gufunc(a, signature=signature)
+    if upper:
+        r = r.T.conj()
```

Comment:
This is only correct for 2-d arrays, while I think the input can be a stack of matrices. Replace with
```
r = matrix_transpose(r).conj()
```

Reply:
Ah, right - fixed!